OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -491,29 +498,49 @@ public void RemoveInfluence()
 
 		public Activity MoveTo(CPos cell, int nearEnough)
 		{
+			var moveWrapper = self.TraitOrDefault<IWrapMove>();
```

Comment:
We can reduce the duplication and trait queries by doing something like:
```csharp
// Lives up with the other trait references and assigned from Created using self.TraitsImplementing<IWrapMove>().ToArray();
IWrapMove[] moveWrappers;

Activity WrapMove(Func<Actor, Activity> inner)
{
	var moveWrapper = moveWrappers.FirstOrDefault(Exts.IsTraitEnabled);
	if (moveWrapper != null)
		return moveWrapper.WrapMove(inner);

	return inner(self);
}

public Activity MoveTo(CPos cell, int nearEnough)
{
	return WrapMove(self => new Move(self, cell, WDist.FromCells(nearEnough)));
}

// ... and so on for the others
```

... but see my question below, as this may change if we want to change the way the wrapping works.

Reply:
Deduplicated the code as suggested.